repos/system_upgrade/common/configs/rhui.py

```diff
@@ -0,0 +1,126 @@
+"""
+Configuration keys for RHUI.
+
+In case of RHUI in private regions it usual that publicly known RHUI data
+is not valid. In such cases it's possible to provide the correct expected
+RHUI data to correct the in-place upgrade process.
+"""
+
+from leapp.actors.config import Config
+from leapp.models import fields
+
+RHUI_CONFIG_SECTION = 'rhui'
+
+
+# @Note(mhecko): We use to distinguish config instantiated from default values that we should ignore
+# #              Maybe we could make all config values None and detect it that way, but then we cannot
+# #              give the user an example how the config should look like.
+class RhuiUseConfig(Config):
+    section = RHUI_CONFIG_SECTION
+    name = "use_config"
+    type_ = fields.Boolean(default=False)
+    default = False
+    description = """
+        Use values provided in the configuration file to override leapp's decisions.
+    """
+
+
+class RhuiSourcePkgs(Config):
+    section = RHUI_CONFIG_SECTION
+    name = "source_clients"
+    type_ = fields.List(fields.String())
+    default = []
+    description = """
+        The name of the source RHUI client RPMs (to be removed from the system).
+        Default: rhui
+    """
+
+
+class RhuiTargetPkgs(Config):
+    section = RHUI_CONFIG_SECTION
+    name = "target_clients"
+    type_ = fields.List(fields.String())
+    default = []
+    description = """
+        The name of the target RHUI client RPM (to be installed on the system).
+        Default: rhui
+    """
+
+
+class RhuiCloudProvider(Config):
+    section = RHUI_CONFIG_SECTION
+    name = "cloud_provider"
+    type_ = fields.String(default="rhui")
+    default = ""
+    description = """
+        Cloud provider name that should be used internally by leapp.
+
+        Leapp recognizes the following cloud providers:
+            - azure
+            - aws
+            - google
+
+        Cloud provider information is used for triggering some provider-specific modifications. The value also
+        influences how leapp determines target repositories to enable.
```

Comment:
In the case of RHUI config, there are no defaults - everything should be empty by default so that we can detect and require all of the config values to be provided by the user. The reason for no defaults is tha we cannot provide any reasonable ones - almost everything is different from provider to provider.

Reply:
@MichalHe that makes sense.